lms/djangoapps/badges/backends/badgr.py

```diff
@@ -166,12 +179,96 @@ def _create_assertion(self, badge_class, user, evidence_url):
         self._send_assertion_created_event(user, assertion)
         return assertion
 
-    @staticmethod
-    def _get_headers():
+    def _encrypt_token(self, token):
+        """
+        Use the BADGR_TOKENS_ENCRYPT_KEY to encrypt a token.
```

Comment:
Using the `SECRET_KEY` sounds better yes, thanks! I've updated the PR.

Reply:
On second thought, my logic is incorrect in case `SECRET_KEY` contains multibyte characters. 

For example, setting `SECRET_KEY` to `"I❤️ Python"` would lead to

```
ValueError: Fernet key must be 32 url-safe base64-encoded bytes.
```

So, instead of

```python
fernet_key = base64.urlsafe_b64encode(
    settings.SECRET_KEY[:32].ljust(32).encode('utf-8')
)
f = Fernet(fernet_key)
```

you'll want something like

```python
fernet_key = base64.urlsafe_b64encode(
    settings.SECRET_KEY.ljust(64).encode('utf-8')[:32]
)
f = Fernet(fernet_key)
```
... in other words, pad, then encode to bytes, then take the first 32, then base64-encode that.